File: LSP_ThirdWheel/LSP_ThirdWheel/VisualBasic/LSPTests.vb
<TestClass()>
Public Class LSPTests
    <TestMethod()>
    Public Sub Square_SetHeight_DoesNotAffectWidth()
        'arrange
        Dim square = New Square()

        'act and assert
        SetHeight_DoesNotAffectWidth(square)
    End Sub

    <TestMethod()>
    Public Sub Rectangle_SetHeight_DoesNotAffectWidth()
        'arrange
        Dim rectangle = New Rectangle()

        'act and assert
        SetHeight_DoesNotAffectWidth(rectangle)
    End Sub

    Public Sub SetHeight_DoesNotAffectWidth(rectangle As Rectangle)
        'arrange
        Dim expectedWidth = 4
        rectangle.Width = 4

        'act
        rectangle.Height = 7

        'assert
        Assert.AreEqual(expectedWidth, rectangle.Width)
    End Sub
End Class

File: LSP_ThirdWheel/LSP_ThirdWheel/CSharp/LSPTests.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharp
{
    [TestClass]
    public class LSPTests
    {
        [TestMethod]
        public void Square_SetHeight_DoesNotAffectWidth()
        {
            //arrange
            var square = new Square();

            //act and assert
            SetHeight_DoesNotAffectWidth(square);
        }

        [TestMethod]
        public void Rectangle_SetHeight_DoesNotAffectWidth()
        {
            //arrange
            var rectangle = new Rectangle();

            //act and assert
            SetHeight_DoesNotAffectWidth(rectangle);
        }

        public void SetHeight_DoesNotAffectWidth(Rectangle rectangle)
        {
            //arrange
            var expectedWidth = 4;
            rectangle.Width = expectedWidth;

            //act
            rectangle.Height = 7;

            //assert
            Assert.AreEqual(expectedWidth, rectangle.Width);
        }
    }
}
